socketio/src/client/builder.rs

```diff
@@ -172,6 +174,27 @@ impl ClientBuilder {
         self
     }
 
+    /// Sets data sent in the opening request.
+    /// # Example
+    /// ```rust
+    /// use rust_socketio::{ClientBuilder};
+    /// use serde_json::json;
+    ///
+    /// let socket = ClientBuilder::new("http://localhost:4204/")
+    ///     .namespace("/admin")
+    ///     .auth(json!({ "password": "1337" }))
+    ///     .on("error", |err, _| eprintln!("Error: {:#?}", err))
+    ///     .connect();
+    ///
+    /// ```
+    pub fn auth<T: TryInto<serde_json::Value>>(mut self, auth: T) -> Self {
+        self.auth = match auth.try_into() {
+            Ok(value) => Some(value),
+            Err(_) => None,
```

Comment:
In this case we are discarding the error and don't tell the user about it, which is not good (imagine a situation where you need to debug the client and realize this is the bug :D). I think it's better to return a ´Result<Self>` here and map the error to our own socket.io error (´Error::InvalidJson` respectively).

Reply:
Ahhh my bad! `serde_json::Value` does not provide a `Try_From<String>` implementation. In that case `T: Into<serde_json::Value>` would be the best. Sorry for the confusion!